fundingmanager_test.go

```diff
@@ -0,0 +1,596 @@
+	if err != nil {
+		t.Fatalf("failed creating fundingManager: %v", err)
+	}
+	if err = aliceFundingMgr.Start(); err != nil {
+		t.Fatalf("failed starting fundingManager: %v", err)
+	}
+
+	bobTestDir, err = ioutil.TempDir("", "boblnwallet")
+	if err != nil {
+		t.Fatalf("unable to create temp directory: %v", err)
+	}
+	bobMsgChan = make(chan lnwire.Message)
+	bobAnnounceChan = make(chan lnwire.Message)
+	bobFundingMgr, err = createTestFundingManager(t, bobPubKey, bobTestDir, bobPrivKeyBytes[:], netParams, rpcConfig, miningNode, chainNotifier, bobMsgChan, bobAnnounceChan)
+	if err != nil {
+		t.Fatalf("failed creating fundingManager: %v", err)
+	}
+	if err = bobFundingMgr.Start(); err != nil {
+		t.Fatalf("failed starting fundingManager: %v", err)
+	}
+}
+
+func tearDownFundingManagers() {
+	os.RemoveAll(bobTestDir)
+	os.RemoveAll(aliceTestDir)
+	miningNode.TearDown()
+}
+
+func TestFundingManager_normalWorkflow(t *testing.T) {
+	setupFundingManagers(t)
+	defer tearDownFundingManagers()
+
+	// Create a funding request and start the workflow
+	errChan := make(chan error, 1)
+	updateChan := make(chan *lnrpc.OpenStatusUpdate, 1)
+	initReq := &openChanReq{
+		targetPeerID:    int32(1),
+		targetPubkey:    bobPubKey,
+		localFundingAmt: 500000,
+		pushAmt:         0,
+		numConfs:        1,
+		updates:         updateChan,
+		err:             errChan,
+	}
+
+	aliceFundingMgr.initFundingWorkflow(bobAddr, initReq)
+
+	// Alice should have sent the init message to Bob
+	fundingReq := <-aliceMsgChan
+	if fundingReq.MsgType() != lnwire.MsgSingleFundingRequest {
+		t.Fatalf("expected MsgSingleFundingRequest to be sent from alice")
+	}
+
+	// Let Bob handle the init message
+	bobFundingMgr.processFundingRequest(fundingReq.(*lnwire.SingleFundingRequest), aliceAddr)
+
+	// and Bob should answer with a fundingResponse
+	fundingResponse := <-bobMsgChan
+	if fundingResponse.MsgType() != lnwire.MsgSingleFundingResponse {
+		t.Fatalf("expected MsgSingleFundingResponse to be sent from bob")
+	}
+
+	// forward response to Alice
+	aliceFundingMgr.processFundingResponse(fundingResponse.(*lnwire.SingleFundingResponse), bobAddr)
+
+	// Alice respond with a FundingComplete messages
+	fundingComplete := <-aliceMsgChan
+	if fundingComplete.MsgType() != lnwire.MsgSingleFundingComplete {
+		t.Fatalf("expected MsgSingleFundingComplete to be sent from alice")
+	}
+
+	// give it to Bob
+	bobFundingMgr.processFundingComplete(fundingComplete.(*lnwire.SingleFundingComplete), aliceAddr)
+
+	// Finally, Bob should send the SingleFundingSignComplete message
+	fundinSignComplete := <-bobMsgChan
+	if fundinSignComplete.MsgType() != lnwire.MsgSingleFundingSignComplete {
+		t.Fatalf("expected MsgSingleFundingSignComplete to be sent from bob")
+	}
+
+	// forward signature to Alice
+	aliceFundingMgr.processFundingSignComplete(fundinSignComplete.(*lnwire.SingleFundingSignComplete), bobAddr)
+
+	// Small sleep to make sure the transaction is broadcast
```

Comment:
Rather than using a sleep here, the `ChainNotifier` instance can be used in order to make this test a bit more snappy.